cmd/metrics.go

```diff
@@ -81,121 +85,195 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
-		prometheus.NewDesc(
-			prometheus.BuildFQName("minio", "network", "sent_bytes_total"),
-			"Total number of bytes sent by current MinIO server instance",
-			nil, nil),
-		prometheus.CounterValue,
-		float64(globalConnStats.getTotalOutputBytes()),
-	)
-	ch <- prometheus.MustNewConstMetric(
-		prometheus.NewDesc(
-			prometheus.BuildFQName("minio", "network", "received_bytes_total"),
-			"Total number of bytes received by current MinIO server instance",
-			nil, nil),
-		prometheus.CounterValue,
-		float64(globalConnStats.getTotalInputBytes()),
-	)
+	for _, endpoint := range globalEndpoints {
+		sAPI, err := newStorageAPI(endpoint)
+		if err != nil {
+			continue
+		}
+		dInfo, dErr := sAPI.DiskInfo()
+		if dErr != nil {
+			continue
+		}
+		used := humanize.IBytes(dInfo.Total - dInfo.Free)
+		unit, err := strconv.ParseFloat(strings.Fields(used)[0], 64)
+		if err != nil {
+			continue
+		}
 
-	// Expose cache stats only if available
-	cacheObjLayer := newCacheObjectsFn()
-	if cacheObjLayer != nil {
-		cs := cacheObjLayer.StorageInfo(context.Background())
+		// Total disk usage by the disk
 		ch <- prometheus.MustNewConstMetric(
 			prometheus.NewDesc(
-				prometheus.BuildFQName("minio", "disk", "cache_storage_bytes"),
-				"Total cache capacity on current MinIO server instance",
-				nil, nil),
+				prometheus.BuildFQName("minio", "disk", "storage_used"),
+				"Total disk storage used on the disk",
+				[]string{"mountpoint", "unit"}, nil),
 			prometheus.GaugeValue,
-			float64(cs.Total),
+			unit,
+			endpoint.String(),
+			strings.Fields(used)[1],
 		)
+
+		available := humanize.IBytes(dInfo.Free)
+		unit, err = strconv.ParseFloat(strings.Fields(available)[0], 64)
+		if err != nil {
+			continue
+		}
+
+		// Total available space in the disk
 		ch <- prometheus.MustNewConstMetric(
 			prometheus.NewDesc(
-				prometheus.BuildFQName("minio", "disk", "cache_storage_free_bytes"),
-				"Total cache available on current MinIO server instance",
-				nil, nil),
+				prometheus.BuildFQName("minio", "disk", "storage_available"),
+				"Total available space left on the disk",
+				[]string{"mountpoint", "unit"}, nil),
 			prometheus.GaugeValue,
-			float64(cs.Free),
+			unit,
+			endpoint.String(),
+			strings.Fields(available)[1],
 		)
-	}
 
-	// Expose disk stats only if applicable
-
-	// Fetch disk space info
-	objLayer := newObjectLayerFn()
-	// Service not initialized yet
-	if objLayer == nil {
-		return
-	}
+		total := humanize.IBytes(dInfo.Total)
+		unit, err = strconv.ParseFloat(strings.Fields(total)[0], 64)
+		if err != nil {
+			continue
+		}
 
-	s := objLayer.StorageInfo(context.Background())
+		// Total storage space of the disk
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(
+				prometheus.BuildFQName("minio", "disk", "storage_total"),
+				"Total space on the disk",
+				[]string{"mountpoint", "unit"}, nil),
+			prometheus.GaugeValue,
+			unit,
+			endpoint.String(),
+			strings.Fields(total)[1],
```

Comment:
We should avoid doing strings.Fields() repeatedly, this can increase memory usage.